app/controllers/auth_source_ldaps_controller.rb

```diff
@@ -36,4 +36,14 @@ def destroy
       process_error
     end
   end
+
+  def test_connection
+    temp_auth_source_ldap = AuthSourceLdap.new(params[:auth_source_ldap])
+    begin
+      msg = temp_auth_source_ldap.test_connection
+    rescue Foreman::Exception => exception
+      render :json => exception.message, :status => :internal_server_error and return
```

Comment:
a) Also log the exception with Foreman::Logging.exception, so the details are available in the server log file.
b) Unsure if internal server error is the right response, since we won't be able to tell the different between a real internal error and an error with the LDAP/user input, perhaps :unprocessable_entity would be consistent with the API then.